main/src/ca/uwaterloo/flix/language/phase/Regions.scala

```diff
@@ -35,20 +35,17 @@ import scala.collection.immutable.SortedSet
 object Regions {
 
   def run(root: Root)(implicit flix: Flix): Validation[Unit, CompilationMessage] = flix.phase("Regions") {
-    val errs = ParOps.parMap(root.defs)(kv => visitDef(kv._2)).flatten
+    val errors = ParOps.parMap(root.defs)(kv => visitDef(kv._2)).flatten
 
     // TODO: Instances
-
-    errs match {
-      case Nil => Validation.success(())
-      case es => Validation.SoftFailure((), Chain.from(es))
-    }
+    Validation.toSuccessOrSoftFailure((), errors.asInstanceOf[Iterable[Recoverable]])
+      .asInstanceOf[Validation[Unit, CompilationMessage]]
```

Comment:
But now-- can we remove these instanceof? :)

Reply:
Nope :)

Reply:
Scala can't infer the types :/

Reply:
:(

Reply:
Can we just annotate `errors: List[TypeError.RegionVarEscapes]`?

Reply:
Does not work either

Reply:
Hmm. But in the other phases we do not need casts. Take Safety for example:

https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/phase/Safety.scala#L45

Reply:
Ah but that's because `SafetyError` extends `Recoverable` directly in the trait. `TypeError` can be both `Unrecoverable` and `Recoverable` so each case class extends one of them. We can either introduce a new `RegionErorr` type that extends `Recoverable` or lose some safety by extending both traits directly on `TypeError`. I prefer the former solution.

Reply:
Okay forget everything I said. We forgot to extends either `Recoverable` on `RegionVarEscapes`.